### PR TITLE
Port Python tests workflow to uv

### PR DESCRIPTION
**Pull request by pamelafox**
## Purpose

`uv pip install` is generally 10x faster than normal pip, so this should speed up our CI workflows.

## Does this introduce a breaking change?

When developers merge from main and run the server, azd up, or azd deploy, will this produce an error?
If you're not sure, try it out on an old environment.

```
[ ] Yes
[X] No
```

## Does this require changes to learn.microsoft.com docs?

This repository is referenced by [this tutorial](https://learn.microsoft.com/azure/developer/python/get-started-app-chat-template)
which includes deployment, settings and usage instructions. If text or screenshot need to change in the tutorial,
check the box below and notify the tutorial author. A Microsoft employee can do this for you if you're an external contributor.

```
[ ] Yes
[X] No
```

## Type of change

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[X] Other... Please describe:
```

## Code quality checklist

See [CONTRIBUTING.md](https://github.com/Azure-Samples/azure-search-openai-demo/blob/main/CONTRIBUTING.md#submit-pr) for more details.

- [ ] The current tests all pass (`python -m pytest`).
- [ ] I added tests that prove my fix is effective or that my feature works
- [ ] I ran `python -m pytest --cov` to verify 100% coverage of added lines
- [ ] I ran `python -m mypy` to check for type errors
- [ ] I either used the pre-commit hooks or ran `ruff` and `black` manually on my code.

### CODE REVIEW

File: .github/workflows/python-test.yaml [hunk 1]
```diff
@@ -26,13 +26,21 @@ jobs:
...
+      UV_SYSTEM_PYTHON: 1
     steps:
         - uses: actions/checkout@v4
         - name: Setup python
           uses: actions/setup-python@v5
           with:
             python-version: ${{ matrix.python_version }}
             architecture: x64
+        - name: Install uv
+          uses: astral-sh/setup-uv@v3
```
Comment (mattgotteiner) [inline .github/workflows/python-test.yaml:39]: This is pretty cool - where can I learn more about UV?

File: app/frontend/src/assets/github.svg [hunk 2]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+<svg width="98" height="96" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" clip-rule="evenodd" d="M48.854 0C21.839 0 0 22 0 49.217c0 21.756 13.993 40.172 33.405 46.69 2.427.49 3.316-1.059 3.316-2.362 0-1.141-.08-5.052-.08-9.127-13.59 2.934-16.42-5.867-16.42-5.867-2.184-5.704-5.42-7.17-5.42-7.17-4.448-3.015.324-3.015.324-3.015 4.934.326 7.523 5.052 7.523 5.052 4.367 7.496 11.404 5.378 14.235 4.074.404-3.178 1.699-5.378 3.074-6.6-10.839-1.141-22.243-5.378-22.243-24.283 0-5.378 1.94-9.778 5.014-13.2-.485-1.222-2.184-6.275.486-13.038 0 0 4.125-1.304 13.426 5.052a46.97 46.97 0 0 1 12.214-1.63c4.125 0 8.33.571 12.213 1.63 9.302-6.356 13.427-5.052 13.427-5.052 2.67 6.763.97 11.816.485 13.038 3.155 3.422 5.015 7.822 5.015 13.2 0 18.905-11.404 23.06-22.324 24.283 1.78 1.548 3.316 4.481 3.316 9.126 0 6.6-.08 11.897-.08 13.526 0 1.304.89 2.853 3.316 2.364 19.412-6.52 33.405-24.935 33.405-46.691C97.707 22 75.788 0 48.854 0z" fill="#fff"/></svg>
```
Comment (mattgotteiner) [inline app/frontend/src/assets/github.svg:1]: This is just a newline change?

**Review by mattgotteiner — COMMENTED** [review]
I think a few files have newline only changes - other than that this is great!

**Comment by pamelafox** [comment]
Oops, this PR had my precommit changes too. I'll update from main after merging that one.

File: .github/workflows/python-test.yaml (hunk 1 above)
Comment (pamelafox) [reply .github/workflows/python-test.yaml:39]: From my old colleague Charlie Marsh who started it all!
https://talkpython.fm/episodes/show/476/unified-python-packaging-with-uv
Or just the docs:
https://github.com/astral-sh/uv

**Pull request merged**